File: doc-examples/mongo-example-records-java/src/main/java/example/Manufacturer.java

package example;

import org.jspecify.annotations.Nullable;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import org.bson.types.ObjectId;

@MappedEntity
public record Manufacturer(
        @Id
        @GeneratedValue
        @Nullable
        ObjectId id,
        String name) {
}

File: doc-examples/mongo-example-kotlin-ksp/src/main/kotlin/example/Manufacturer.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import org.bson.types.ObjectId

@MappedEntity
data class Manufacturer(
        @field:Id
        @GeneratedValue
        var id: ObjectId?,
        val name: String
)
